api/envoy/config/filter/http/jwt_authn/v2alpha/config.proto

```diff
@@ -183,44 +183,178 @@ message JwtHeader {
   string value_prefix = 2;
 }
 
-// This is the Envoy HTTP filter config for JWT authentication.
-// [#not-implemented-hide:]
-message JwtAuthentication {
-  // List of JWT rules to valide.
-  repeated JwtRule rules = 1;
+// This message specifies a Jwt requirement.
+message RequiredProvider {
+  // Specify the issuer for the required JwtProvider.
+  string issuer = 1;
 
-  // If true, the request is allowed if JWT is missing or JWT verification fails.
-  // Default is false, a request without JWT or failed JWT verification is not allowed.
-  bool allow_missing_or_failed = 2;
+  // This audiences field overrides the one in JwtProvider if not empty.
+  repeated string audiences = 2;
+}
 
-  // This field lists the patterns allowed to bypass JWT verification. This only applies when
-  // `allow_missing_or_failed_jwt` is false. Under this config, if a request doesn't have JWT, it
-  // will be rejected. But some requests still needed to be forwarded without JWT, such as OPTIONS
-  // for CORS and some health checking paths.
+// This message specifies a Jwt requirement. It consists of multiple "or_jwt", each "or_jwt"
+// consists of multiple "and_jwt". Each "and_jwt" specifies a required provider.
+// If the message is empty, Jwt is not required.
+//
+// .. code-block:: yaml
+//
+//   or_jwt:
+//      - and_jwt:
+//         - issuer: issuer1
+//         - issuer: issuer2
+//           audiences: [audience1, audience2]
+//      - and_jwt:
+//         - issuer: issuer3
+//
+// Above example specifies this requirement: (issuer1 & issuer2) || (issuer3)
+//
+// Route.per_filter_config for jwt_authn filter uses this message.
+message JwtRequirement {
+  // Specify list of required JwtProviders. Their results are AND-ed;
+  // all of them must pass, if one of them fails or missing, it fails.
+  message AndRequirement {
+    repeated RequiredProvider and_jwt = 1;
```

Comment:
add validate rule that this must not be empty.

I feel there should be better name than `and_jwt` / `or_jwt`...

Reply:
Added validate rule, and renamed to "one_of" and "all_of"